### PR TITLE
Merge kubevirt/libvirt into kubevirt/kubevirt

### PR DESCRIPTION
**Pull request by rmohr**
**What this PR does / why we need it**:

Centralize our dependency management by using https://github.com/rmohr/bazeldnf to resolve our RPMs on a bazel level.

This allows us to have only `bazel` and ` C-toolchain as an external dependency. All shared libraries can now consumed fully through bazel.

Benefits are:
 * Full transparent tracking of changed RPM dependencies
 * Managing all dependencies in one repo and no longer requiring kubevirt/libvirt to pull in new qemu/libvirt version:
   - allows easy replacement and testing with e.g. latest libvirt/qemu with out pre-pushing them.
 * almost no need anymore to ever update the build-mage which allows us to pull it out to kubevirtci.
   - no more docker-in-docker runs needed in CI
   -  streamlined developer experience, by being able to express all changes in a single PR without flow-interrupts
 * All RPMs can are cached and mirrored
   - allows transparent and easy moving between versions (rollback or modifications can be done and tracked easily) 
 * Easy to extend for multi-arch support
 * Fixes as a side-effect the need to build as root in order to work properly with RPMs like we did so far

Run

```
make rpm-deps
```

to update RPM dependencies to the latest state.

Modify `repo.yaml` to use different fedora or copr versions. Initially it looks like this:

```yaml
repositories:
- arch: x86_64
  metalink: https://mirrors.fedoraproject.org/metalink?repo=fedora-32&arch=x86_64
  name: 32-x86_64-primary-repo
- arch: x86_64
  metalink: https://mirrors.fedoraproject.org/metalink?repo=updates-released-f32&arch=x86_64
  name: 32-x86_64-update-repo
- arch: x86_64
  baseurl: https://download.copr.fedorainfracloud.org/results/@kubevirt/libvirt-6.6.0-8.el8/fedora-32-x86_64/
  name: kubevirt/libvirt-copr-x86_64
- arch: x86_64
  baseurl: https://download.copr.fedorainfracloud.org/results/@kubevirt/qemu-kvm-5.1.0-16.el8/fedora-32-x86_64/
  name: kubevirt/qemu-copr-x86_64
- arch: x86_64
  baseurl: https://download.copr.fedorainfracloud.org/results/@kubevirt/seabios-1.14.0-1.el8/fedora-32-x86_64/
  name: kubevirt/seabios-copr-x86_64
```

Finally a `make verify-rpm-deps` target was added to ensure that all referenced RPMs are signed by known GPG keys. This will be integrated in CI, to allow engineers to locally update and modify RPM dependencies.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

More background can be found at https://github.com/rmohr/bazeldnf.

In short, `bazeldnf` does:
 * introducing a `rpmtree` target, which is not much more than just a bundle of RPMs converted into single tar archives. It can be used for instance as a container-layer, or as a source for required headers or libraries.
 * introcuing a `tar2files` target which can make a selected subset of headers and shared libraries available for `cc_library` rules by taking tar files (like emited from `rpmtree` targets) as input. Finally `cc_library` targets can then be passed to `go` code and used via `cgo`.
 * Finally `bazeldnf` has a sub-command called `ldd` which can resolve transitive dependencies of shared libraries and write them as `tar2files` targets. This ensures linking only required shared libraries.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by rmohr** [comment]
/cc @crobinso
/cc @andreabolognani 

FYI

**Comment by rmohr** [comment]
Ready for reviews.

**Review by davidvossel — APPROVED** [review]
/approve
/lgtm
